src/Raven.Client/ServerWide/DatabaseRecord.cs

```diff
@@ -203,32 +205,118 @@ public void AddIndex(IndexDefinition definition, string source, DateTime created
             {
                 history.RemoveRange(revisionsToKeep, history.Count - revisionsToKeep);
             }
+
+            if (IsRolling(definition.Rolling, globalRollingSetting))
+            {
+                if (differences == null || (differences.Value & IndexDefinitionCompareDifferences.ReIndexRequiredMask) != 0)
+                    InitializeRollingDeployment(definition.Name, createdAt, raftIndex);
+            }
         }
 
         public void AddIndex(AutoIndexDefinition definition)
         {
+            AddIndex(definition, DateTime.UtcNow, 0, false);
+        }
+
+        internal void AddIndex(AutoIndexDefinition definition, DateTime createdAt, long raftIndex, bool globalRollingSetting)
+        {
+            IndexDefinitionCompareDifferences? differences = null;
+
             if (AutoIndexes.TryGetValue(definition.Name, out AutoIndexDefinition existingDefinition))
             {
-                var result = existingDefinition.Compare(definition);
+                differences = existingDefinition.Compare(definition);
 
-                if (result == IndexDefinitionCompareDifferences.None)
+                if (differences == IndexDefinitionCompareDifferences.None)
                     return;
 
-                result &= ~IndexDefinitionCompareDifferences.Priority;
-                result &= ~IndexDefinitionCompareDifferences.State;
+                differences &= ~IndexDefinitionCompareDifferences.Priority;
+                differences &= ~IndexDefinitionCompareDifferences.State;
 
-                if (result != IndexDefinitionCompareDifferences.None)
-                    throw new NotSupportedException($"Can not update auto-index: {definition.Name} (compare result: {result})");
+                if (differences != IndexDefinitionCompareDifferences.None)
+                    throw new NotSupportedException($"Can not update auto-index: {definition.Name} (compare result: {differences})");
             }
 
             AutoIndexes[definition.Name] = definition;
+            
+            if (IsRolling(definition.Rolling, globalRollingSetting))
+            {
+                if (differences == null || (differences.Value & IndexDefinitionCompareDifferences.ReIndexRequiredMask) != 0)
+                    InitializeRollingDeployment(definition.Name, createdAt, raftIndex);
+            }
+        }
+
+        internal static bool IsRolling(bool? fromDefinition, bool fromSetting)
+        {
+            if (fromDefinition == false)
+                return false;
+
+            return fromSetting || fromDefinition == true;
```

Comment:
We can set rolling globally or per index definition, which is nullable.
If the definition is set (not null) it override any global setting.
otherwise we use the global setting.